src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -426,10 +429,12 @@ def get_access_token_for_resource(self, username, tenant, resource):
         return access_token
 
     @staticmethod
-    def _try_parse_msi_account_name(subscription_name):
-        parts = subscription_name.split('-', 1)
-        if parts[0] in MsiAccountTypes.valid_msi_account_types():
-            return parts[0], (None if len(parts) <= 1 else parts[1])
+    def _try_parse_msi_account_name(account):
+        subscription_name, user = account[_SUBSCRIPTION_NAME], account[_USER_ENTITY].get(_USER_NAME)
+        if user in [_SYSTEM_ASSIGNED_IDENTITY, _USER_ASSIGNED_IDENTITY]:
```

Comment:
For service principals, could `user` be `None`?

Reply:
The `user` section and the user name inside will always be there regardless the account type. For service principal, it will be the client id. So we should be fine.